src/NodaTime.Demo/OffsetDateTimeDemo.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright 2019 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NUnit.Framework;
+
+namespace NodaTime.Demo
+{    
+    public class OffsetDateTimeDemo
+    {
+        [Test]
+        public void Construction()
+        {
+            LocalDateTime localDateTime = new LocalDateTime(1985, 10, 26, 1, 18);
+            Offset offset = Offset.FromHours(-5);
+            OffsetDateTime offsetDateTime = Snippet.For(new OffsetDateTime(localDateTime, offset));
+
+            Assert.AreEqual(localDateTime, offsetDateTime.LocalDateTime);
+            Assert.AreEqual(offset, offsetDateTime.Offset);
+        }
+
+        [Test]
+        public void WithCalendar()
+        {
+            LocalDateTime localDateTime = new LocalDateTime(1985, 10, 26, 1, 18, CalendarSystem.Gregorian);
+            Offset offset = Offset.FromHours(-5);
+            OffsetDateTime original = new OffsetDateTime(localDateTime, offset);
+            OffsetDateTime updated = Snippet.For(original.WithCalendar(CalendarSystem.Iso));
+
+            Assert.AreNotEqual(original.LocalDateTime, updated.LocalDateTime);
+            Assert.AreEqual(original.Offset, updated.Offset);
+            Assert.AreEqual(CalendarSystem.Iso, updated.Calendar);
+        }
+
+        [Test]
+        public void WithOffset()
+        {
+            LocalDateTime localDateTime = new LocalDateTime(1985, 10, 26, 1, 18);
+            Offset offset = Offset.FromHours(-5);
+            OffsetDateTime original = new OffsetDateTime(localDateTime, offset);
+            OffsetDateTime updated = Snippet.For(original.WithOffset(Offset.FromHours(-3)));
+
+            Assert.AreNotEqual(original.LocalDateTime, updated.LocalDateTime);
+            Assert.AreEqual(Offset.FromHours(-3), updated.Offset);
+        }
+
+        [Test]
+        public void WithDateAdjuster()
+        {
+            LocalDateTime localDateTime = new LocalDateTime(1985, 10, 26, 1, 18);
+            Offset offset = Offset.FromHours(-5);
+            OffsetDateTime original = new OffsetDateTime(localDateTime, offset);
+            OffsetDateTime updated = Snippet.For(original.With(x => x.PlusYears(30)));
+
+            Assert.AreEqual(
+                new LocalDateTime(2015, 10, 26, 1, 18), 
+                updated.LocalDateTime);
+            Assert.AreEqual(original.Offset, updated.Offset);
+        }
+
+        [Test]
+        public void WithTimeAdjuster()
+        {
+            LocalDateTime localDateTime = new LocalDateTime(1985, 10, 26, 1, 18);
+            Offset offset = Offset.FromHours(-5);
+            OffsetDateTime original = new OffsetDateTime(localDateTime, offset);
+            OffsetDateTime updated = Snippet.For(original.With(x => x.PlusHours(1)));
```

Comment:
This is probably not a good adjuster to use - it suggests it adds an hour to the LocalDateTime, when it doesn't really. (If you set the time to 23:30 you'll see what I mean.)

I'd use the `TimeAdjusters.AtMidnight` adjuster

Reply:
I can't find this adjuster. It doesn't look like it's located at the TimeAdjusters class at all. Is it in some special namespace or something?

Reply:
Whoops - my mistake. Try TruncateToHour or similar.